### PR TITLE
readme: fix spelling and clarity in README

### PR DESCRIPTION
**Pull request by dbkinder**
Fixed some spelling and grammar issues, and made clarity edits.

Added link to building without docker section.

Clarified only linux build environment supported.

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -2,21 +2,28 @@ intel-iot-refkit
 ################
 
 This README file contains information on setting up, building, and booting
-Iot Reference OS Kit for Intel(r) Architecture project.
+the IoT Reference OS Kit for Intel(r) Architecture project.
+
+.. note::
+
+   Only Linux development hosts are currently support, as documented
```
Comment (mythi) [inline README.rst]: supported

File: README.rst [hunk 2]
```diff
@@ -109,7 +119,7 @@ https://git-scm.com/docs/git-submodule
 Installing the Images
 =====================
 
-See detailed instructions in doc/howtos/image-install.rst.
+See detailed instructions in `how to install images <doc/howtos/image-install.rst>`__.
```
Comment (gvancuts) [inline README.rst]: Just curious, why do you need two trailing underscore characters?

File: README.rst (hunk 1 above)
Comment (dbkinder) [inline README.rst]: sigh... I shouldn't do these patches when I'm distracted.  Sorry about the spelling error.

File: README.rst (hunk 2 above)
Comment (dbkinder) [inline README.rst]: I was experimenting with "anonymous hyperlinks" (where the reference does not create a lingering name that others could reference too).  In this case it doesn't make a difference so I put it back to the regular single underscore.  While this hyperlinking will work on GitHub, this format of hyperlinking would be a problem if we were using Sphinx, but for now it does give us the ability to link between .rst files in GitHub (normally we'd be linking to the generated .html files and not a .rst file).

Comment (gvancuts) [inline README.rst]: Thanks for the explanation @dbkinder . I OK with two trailing underscores, I just did not know what it meant and I now got educated, thanks :-)

**Comment by mythi** [comment]
Merged in #71. Thanks @dbkinder

**Pull request closed without merging**